pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -124,15 +135,15 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 			MonitorSecret: string(secrets.Data[MonSecretNameKey]),
 			Context:       context,
 		}
-		if cephUsername, ok := secrets.Data[CephUsernameKey]; ok {
+		if cephUsername, ok := secrets.Data[CephLegacyUsernameKey]; ok {
 			clusterInfo.CephCred.Username = string(cephUsername)
-			clusterInfo.CephCred.Secret = string(secrets.Data[CephUserSecretKey])
+			clusterInfo.CephCred.Secret = string(secrets.Data[CephLegacyUserSecretKey])
 		} else if adminSecretKey, ok := secrets.Data[adminSecretNameKey]; ok {
```

Comment:
The code block on lines 181-206 needs to be moved up here. We also can delete these lines 141-149 since it is for a legacy conversion from a few releases ago. So the code flow starting on line 138 would be something like:
```go
// First check for the most common case of the `rookoperator` keyring existing
if cephUsername, ok := secrets.Data[CephUsernameKey]; ok && string(cephUsername) == "client.rookoperator"{
	clusterInfo.CephCred.Username = string(cephUsername)
	clusterInfo.CephCred.Secret = string(secrets.Data[CephUserSecretKey])
} else if isExternal { // **this code block is from lines 181-185**
	secrets.Data[CephUsernameKey] = secrets.Data[CephLegacyUsernameKey]
	secrets.Data[CephUserSecretKey] = secrets.Data[CephLegacyUserSecretKey]
	// we don't want to create client.rookoperator for external cluster
} else {
	// The upgrade path is now triggered since the new key name doesn't exist, or the client.admin keyring is still being used
	// **Move lines 186-205 here**
}
```

Reply:
done 👍🏽